src/lib/mdns/Resolver.h

```diff
@@ -43,18 +49,32 @@ struct ResolvedNodeData
 #endif // CHIP_PROGRESS_LOGGING
     }
 
+    Optional<uint32_t> GetMrpRetryIntervalIdle() const
+    {
+        return mMrpRetryIntervalIdle != kUndefinedRetryInterval ? Optional<uint32_t>{ mMrpRetryIntervalIdle }
+                                                                : Optional<uint32_t>{};
+    }
+
+    Optional<uint32_t> GetMrpRetryIntervalActive() const
+    {
+        return mMrpRetryIntervalActive != kUndefinedRetryInterval ? Optional<uint32_t>{ mMrpRetryIntervalActive }
+                                                                  : Optional<uint32_t>{};
+    }
+
     PeerId mPeerId;
-    Inet::InterfaceId mInterfaceId;
-    Inet::IPAddress mAddress;
-    uint16_t mPort;
+    Inet::IPAddress mAddress             = Inet::IPAddress::Any;
+    Inet::InterfaceId mInterfaceId       = INET_NULL_INTERFACEID;
+    uint16_t mPort                       = 0;
+    char mHostName[kMaxHostNameSize + 1] = {};
+    bool mSupportsTcp                    = false;
+    uint32_t mMrpRetryIntervalIdle       = kUndefinedRetryInterval;
```

Comment:
Should consider making these private so consumers go through the accessors...  Separate PR for that, please.

Reply:
@Damian-Nordic see above.